src/utils/date.ts

```diff
@@ -10,4 +11,127 @@ export function isDate(dateString?: ValueType) {
+ * @returns string if stringFormat is passed else a date object
+ */
+export const formatDate = (
+  date: string | Date,
+  { asTimestamp, stringFormat, suffix }: formatDateOptions = {}
+) => {
+  const localDate = dayjs(date).local();
+  if (asTimestamp) return localDate.valueOf().toString();
+  const formattedDate = stringFormat
+    ? localDate.format(stringFormat)
+    : localDate;
+  return suffix ? `${formattedDate}${suffix}` : formattedDate;
+};
+
+/**
+ * Parse the given date relative to the user locale
+ *
+ * @param date
+ * @param fromNow give relative date from now
+ * @returns relative date as string
+ */
+export const relativeDateTime = (date: string | Date, fromNow = false) => {
+  const t = dayjs.utc(date).local();
+  const n = dayjs();
+
+  if (n.isSame(t, "day") || fromNow) {
+    return t.fromNow();
+  }
+
+  if (n.diff(t, "days") === 1) {
+    return `${t.format(DATE_FORMATS.TIME)} yesterday`;
+  }
+
+  return t.format(DATE_FORMATS.LONG);
+};
+
+/**
+ *
+ * @param value date to check
+ * @returns boolean check if value is a valid date
+ */
+export const isValidDate = (value: DateConfig) => dayjs(value).isValid();
+
+/**
+ *
+ * @param value date to convert
+ * @returns string of ISO date
+ */
+export const dateToISOString = (value: DateConfig) =>
+  dayjs(value).toISOString();
+
+/**
+ *
+ * @param value date to convert
+ * @returns Native date object
+ */
+export const dateToJSDate = (value: DateConfig) => dayjs(value).toDate();
+
+/**
+ *
+ * @param value date to format
+ * @param format custom string format for the date
+ * @returns formatted string of the date
+ */
+export const customDateFormattor = (value: DateConfig, format: string) =>
+  dayjs(value).format(format);
+
+/**
+ *
+ * @param decValue Time in number needed to be decreased from now
+ * @param decUnit Unit of time to be decreased. Must be of type ManipulateType from dayjs
+ * @returns Day.js object with reduced time
+ */
+export const subtractDateFromNow = (decValue: number, decUnit: string) =>
+  dayjs().subtract(decValue, decUnit as ManipulateType);
+
+/**
+ *
+ * @param date1 First date
+ * @param date2 Second Date
+ * @param diffUnit To get the difference in another unit of measurement, pass that measurement as the second argument.
+ * @param isDiffInFloat Boolean value to specify if difference is needed in decimals.
+ * @returns Difference between the first and second date in milliseconds (or required units if diffUnit is specified)
+ */
+export const dateDiff = (
+  date1: DateConfig,
+  date2: DateConfig,
+  diffUnit?: string,
+  isDiffInFloat?: boolean
+) => dayjs(date1).diff(dayjs(date2), diffUnit as ManipulateType, isDiffInFloat);
+
+/**
+ *
+ * @param date1 First date
+ * @param date2 Second Date
+ * @param withoutSuffix If suffix needed at the end of time passed.
+ * @returns a string mentioning the time passed from date2 relative to date1
+ */
+export const timePassedInWords = (
```

Comment:
how is this different from relativeDateTime?

Reply:
The functionality here is totally different.

1. It takes 2 dates as params, rather than 1 for measuring the time different between the two. This is because the second date is not necessarily "now".
2. relativeDateTime does a custom operation as I have mentioned in https://github.com/flanksource/flanksource-ui/pull/496#discussion_r1006519326. This functionality is very different from what is being done here.

Let me know if there is a need to have some separate functionality here.